test/devicedefender/TestDeviceDefender.cpp

```diff
@@ -86,6 +108,48 @@ TEST_F(TestDeviceDefender, Init)
 {
     /**
      * Simple init test
+     * Inject Mocks via init, verify Thing Name and Interval
+     */
+    ASSERT_EQ(deviceDefender->init(manager, notifier, config), 0);
+    ASSERT_STREQ(deviceDefender->getThingName().c_str(), "thing-name value");
+    ASSERT_EQ(deviceDefender->getInterval(), 300);
+}
+
+TEST_F(TestDeviceDefender, Start)
+{
+    /**
+     * Simple Start test
+     * Inject Report Task, init, and start, verify Thing Name and Interval
+     * Verify calls on Mocks and Assert Start returns 0
      */
+    EXPECT_CALL(*task, StartTask()).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*deviceDefender, createReportTask()).Times(1).WillOnce(Return(task));
+    EXPECT_CALL(*deviceDefender, subscribeToTopicFilter()).Times(1).WillOnce(Return());
+    EXPECT_CALL(*notifier, onEvent(_, _)).Times(1).WillOnce(Return());
+
+    ASSERT_EQ(deviceDefender->init(manager, notifier, config), 0);
+    ASSERT_EQ(deviceDefender->start(), 0);
+    ASSERT_STREQ(deviceDefender->getThingName().c_str(), "thing-name value");
+    ASSERT_EQ(deviceDefender->getInterval(), 300);
+}
+
+TEST_F(TestDeviceDefender, StartStop)
+{
+    /**
+     * Start and Stop test
+     * Inject Report Task, init, and start, verify Thing Name and Interval
+     * Verify calls on Mocks and Assert Start and Stop return 0
+     */
+    EXPECT_CALL(*task, StartTask()).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*deviceDefender, createReportTask()).Times(1).WillOnce(Return(task));
+    EXPECT_CALL(*deviceDefender, subscribeToTopicFilter()).Times(1).WillOnce(Return());
+    EXPECT_CALL(*notifier, onEvent(_, _)).Times(2).WillOnce(Return());
```

Comment:
Is it possible to also verify the exact status like `FEATURE_STARTED` sent in the notification?